### ISSUE TITLE
[Feature] Add liniting to ensure yarn.lock is updated

### ISSUE THREAD

**Issue by Mottie** [department-of-veterans-affairs/va.gov-team#16023]
## Issue Description

When `package.json` is updated, it should include a corresponding update to the `yarn.lock` file. This causes the `additional-linting` and `test_and_build` checks to fail when a new pull request is created; but only after a previous pull request gets merged in without a cooresponding `yarn.lock` update.

This is a request to add a new lint check that tests if the `package.json` was updated, and should include an update to the `yarn.lock`.

## Tasks
- [ ] Add linting rule

## Acceptance Criteria
- [ ] Linting rule prevents a `package.json` update without an update to the `yarn.lock`

---
## How to configure this issue
- [ ] **Attached to a Milestone** (when will this be completed?)
- [ ] **Attached to an Epic** (what body of work is this a part of?)
- [ ] **Labeled with Team** (`product support`, `analytics-insights`, `operations`, `service-design`, `tools-be`, `tools-fe`)
- [ ] **Labeled with Practice Area** (`backend`, `frontend`, `devops`, `design`, `research`, `product`, `ia`, `qa`, `analytics`, `contact center`, `research`, `accessibility`, `content`)
- [ ] **Labeled with Type** (`bug`, `request`, `discovery`, `documentation`, etc.)
